packages/nimble-components/src/select/index.ts

```diff
@@ -655,7 +659,8 @@ export class Select
         // don't call super.selectNextOption as that relies on side-effecty
         // behavior to not select disabled option (which no longer works)
         for (let i = this.selectedIndex + 1; i < this.options.length; i++) {
-            if (!this.options[i]?.disabled) {
+            const listOption = this.options[i]!;
+            if (isNimbleListOption(listOption) && isOptionSelectable(listOption)) {
                 this.selectedIndex = i;
```

Comment:
There are several other places in the file where we programmatically set `this.selectedIndex = ...`. Could you scan them and see if there are other bugs lurking that should also call `isOptionSelectable()`?

I suspect they're mostly extreme corner cases like option attributes changing while the drop down is open, but worth a check.

Reply:
The only other place that seemed possibly relevant is where we determine the initial default selected option. The new behavior is that we will only select a default option that isn't hidden (or `visually-hidden`, but that should never be possible, but it's not a harmful aspect of the general check).